sdk/core/core-http-compat/src/extendedClient.ts

```diff
@@ -50,4 +56,41 @@ export class ExtendedServiceClient extends ServiceClient {
       });
     }
   }
+
+  /**
+   * Compatible send operation request function.
+   *
+   * @param operationArguments - Operation arguments
+   * @param operationSpec - Operation Spec
+   * @returns
+   */
+  async sendOperationRequest<T>(
+    operationArguments: OperationArguments,
+    operationSpec: OperationSpec
+  ): Promise<T> {
+    const userProvidedCallBack: RawResponseCallback | undefined =
+      operationArguments?.options?.onResponse;
+
+    if (!operationArguments.options) operationArguments.options = {};
+    let lastResponse: FullOperationResponse | undefined;
+
+    function onResponse(
+      rawResponse: FullOperationResponse,
+      flatResponse: unknown,
+      error?: unknown
+    ): void {
+      lastResponse = rawResponse;
+      if (userProvidedCallBack) {
+        userProvidedCallBack(rawResponse, flatResponse, error);
+      }
+    }
+
+    operationArguments.options.onResponse = onResponse;
```

Comment:
Not sure how likely this is, but since we are modifying the input options object, if the consumer is re-using a reference to this object, it will perhaps cause trouble, e.g.

```ts
const options: OperationOptions = { onResponse: (rawResponse) => { /* do something */ } };
// will work fine
const result = await client.doOperation(options);
// client callback will not be called
const result2 = await client.doOperation(options);
```

Maybe another good test case to add for completeness?

Reply:
@xirzec I have added one more test case. But, actually, the callback will be invoked for both instances as shown in the new test case

Reply:
Hmm I suppose this makes sense because the override will call the previous override which will in turn call the original function. Could this go a little wild if they re-use the same object over and over (somewhere above 10,000 I think we hit the function call limit.)

I was also thinking about if it was possible to have the wrong result get attached if two calls were made in parallel and their responses arrived out of order, but I think we'd always update `lastResponse` immediately after receiving the correct result, I don't think it'll get confused.

Eh, I don't know how important this is. @willmtemple can you check my sanity when you have a minute?

Reply:
whoops meant to tag @witemple-msft

Reply:
@xirzec As discussed on Friday, I have modified the code to use the spread operator on options. Please review